openedx/features/content_type_gating/services.py

```diff
@@ -30,3 +35,14 @@ def content_type_gate_for_block(self, user, block, course_id):
             if (not access and access.error_code == 'incorrect_user_group'):
                 return access.user_fragment
             return None
+
+    def get_gate_for_masquerading_user(self, user, course, course_id):
```

Comment:
Is there a way to make this logic invisible to consumers of the service? i.e. put this logic inside `content_type_gate_for_block`? (not necessarily directly - like, make this a private method and call it from there would likely be cleanest)

Requiring service consumers to know to call this method is an invitation to bugs. edx-platform does this a lot, where an API's context is externalized onto callers rather than internalized, and it makes the platform harder to understand I think.

Reply:
put everything inside the content type gating service and made this method private (prefixed with _) which i think addresses this comment